src/pages/Autolending/LoanTermDropdown.vue

```diff
@@ -0,0 +1,104 @@
+<template>
+	<div>
+		<h3 class="filter-title">
+			Loan term
+		</h3>
+		<kv-dropdown-rounded v-model="loanTerm">
+			<option value="default">
+				All loan terms
+			</option>
+			<option value="6m-or-less">
+				6 months or less
+			</option>
+			<option value="12m-or-less">
+				12 months or less
+			</option>
+			<option value="18m-or-less">
+				18 months or less
+			</option>
+			<option value="24m-or-less">
+				24 months or less
+			</option>
+		</kv-dropdown-rounded>
+	</div>
+</template>
+
+<script>
+import _get from 'lodash/get';
+import gql from 'graphql-tag';
+import KvDropdownRounded from '@/components/Kv/KvDropdownRounded';
+
+export default {
+	inject: ['apollo'],
+	components: {
+		KvDropdownRounded,
+	},
+	data() {
+		return {
+			loanTerm: 'default',
+		};
+	},
+	apollo: {
+		query: gql`{
+			autolending @client {
+				currentProfile {
+					loanSearchCriteria {
+						filters {
+							lenderTerm {
+								max
+							}
+						}
+					}
+				}
+			}
+		}`,
+		preFetch: true,
+		result({ data }) {
+			const loanTermMax = _get(data, 'autolending.currentProfile.loanSearchCriteria.filters.lenderTerm.max');
+			if (loanTermMax <= 6) {
+				this.loanTerm = '6m-or-less';
+			} else if (loanTermMax > 6 && loanTermMax <= 12) {
```

Comment:
For my own understanding, why do we have to do these range checks? We can only set loanTermMax to 6/12/18/24 can lenderTerm.max be any other values?

Reply:
The api and loan search criteria allows any number to be set for the min or max. The 6/12/18/24 options represent our very specific business rules for this interface only.